lib/routes.js

```diff
@@ -190,55 +219,21 @@ module.exports = async function (app, opts) {
   })
 
   app.post(graphqlPath, {
-    schema: {
-      body: {
-        type: 'object',
-        properties: {
-          query: {
-            type: 'string',
-            description: 'the GraphQL query'
-          },
-          persisted: {
-            type: 'boolean',
-            description: 'flag to indicate if requesting for a persisted query using it\'s hash'
-          },
-          operationName: {
-            type: 'string'
-          },
-          variables: {
-            type: ['object', 'null'],
-            additionalProperties: true
-          }
-        }
-      },
-      response: responseSchema
-    },
+    schema: postSchema(allowBatchedQueries),
     attachValidation: true
   }, async function (request, reply) {
     validationHandler(request.validationError)
 
-    let { query } = request.body
-    const {
-      persisted,
-      variables,
-      operationName
-    } = request.body
-
-    if (persisted || onlyPersisted) {
-      query = persistedQueries[query]
+    if (Array.isArray(request.body) && allowBatchedQueries) {
+      // Batched query
+      return Promise.all(request.body.map(async r =>
```

Comment:
```suggestion
      return Promise.all(request.body.map(r =>
```

This does not need to be async.